server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -254,6 +239,55 @@ public void putSynonymsSet(String resourceName, SynonymRule[] synonymsSet, Actio
         }));
     }
 
+    public void putSynonymRule(String synonymSetId, SynonymRule synonymRule, ActionListener<UpdateSynonymsResult> listener) {
+        checkSynonymSetExists(synonymSetId, listener.delegateFailure((l1, obj) -> {
```

Comment:
I hate the `l1`, `l2` naming. Any other suggestion?

Reply:
`synonymListener` and `indexListener`?